scripts/imgtool/image.py

```diff
@@ -166,11 +166,18 @@ def __init__(self, version=None, header_size=IMAGE_HEADER_SIZE,
         else:
             align_lsb = self.max_align & 0x00ff
             align_msb = (self.max_align & 0xff00) >> 8
-            self.boot_magic = bytes([
-                align_lsb, align_msb, 0x2d, 0xe1,
-                0x5d, 0x29, 0x41, 0x0b,
-                0x8d, 0x77, 0x67, 0x9c,
-                0x11, 0x0f, 0x1f, 0x8a, ])
+            if self.endian == "big" :
+                self.boot_magic = bytes([
+                    align_msb, align_lsb, 0x2d, 0xe1,
+                    0x5d, 0x29, 0x41, 0x0b,
+                    0x8d, 0x77, 0x67, 0x9c,
+                    0x11, 0x0f, 0x1f, 0x8a, ])
+            else :
+                self.boot_magic = bytes([
+                    align_lsb, align_msb, 0x2d, 0xe1,
+                    0x5d, 0x29, 0x41, 0x0b,
+                    0x8d, 0x77, 0x67, 0x9c,
+                    0x11, 0x0f, 0x1f, 0x8a, ])
```

Comment:
It's OK as is, but to avoid the copy pasta in the magic you could also have done something like:

```python
lsb = self.max_align & 0x00ff
msb = (self.max_align & 0xff00) >> 8
align = bytes([msb, lsb]) if self.endian == "big" else bytes([lsb, msb])
self.boot_magic = align + bytes([0x2d, 0xe1,
                                 0x5d, 0x29, 0x41, 0x0b,
                                 0x8d, 0x77, 0x67, 0x9c,
                                 0x11, 0x0f, 0x1f, 0x8a, ])
```

Reply:
Thank you for the feedback, I changed the code according to your suggestion